datalad/ui/progressbars.py

```diff
@@ -32,7 +32,7 @@ def __init__(self, label=None, fill_text=None, total=None, out=None, unit='B', i
         self.total = total
         self.unit = unit
         self.out = out
-        self._current = initial
+        self._current = self._initial = initial
```

Comment:
So `ProgressBarBase.__init__` has an `initial` parameter.  Confusingly, its children that explicitly define parameters, `SilentProgressBar` and `tqdmProgressBar`, do not.  As far as I can see, no caller in our code base actually instantiates a progress bar class with `initial`.

Instead of extending `tqdmProgressBar.__init__` to have an `initial` parameter and updating `ProgressHandler.emit` to pass it in, what about just making `ProgressHandler.emit` pass `initial` in its call to `start()`?

I'd also be in favor of pruning `initial` from `ProgressBarBase.__init__`, though I don't think that needs to be done in this PR.

Reply:
Ah, I think re-reading your first commit message, this might be answered there.  The idea is that now a bare `.start()` allows re-initializing, I think.

Reply:
Still I think having two spots to passing in `initial` (especially given the inconsistency with classes) isn't a good thing.  I'd be in favor of having `.start()` also store the value it receives for `initial` in an attribute.  Then it `.start()` could still be called without any arguments to re-initialize to that value.

Reply:
thanks Kyle, I *hope* 057dcc0 did what you had in mind. Please let me know if not.

Reply:
>  I _hope_ [057dcc0](https://github.com/datalad/datalad/commit/057dcc079d61c0775e75a20dabbc84d156c2d832) did what you had in mind.

Hmm, I think that makes sense to do, but the core change I was suggesting with "Instead of extending ..."  was something like this (untested):

<details>
<summary>diff</summary>

```diff
diff --git a/datalad/log.py b/datalad/log.py
index 29e5389fc0..414564eeba 100644
--- a/datalad/log.py
+++ b/datalad/log.py
@@ -214,9 +214,8 @@ def emit(self, record):
                 label=getattr(record, 'dlm_progress_label', ''),
                 unit=getattr(record, 'dlm_progress_unit', ''),
                 total=getattr(record, 'dlm_progress_total', None,),
-                initial=getattr(record, 'dlm_progress_initial', 0)
             )
-            pbar.start()
+            pbar.start(initial=getattr(record, 'dlm_progress_initial', 0))
             self.pbars[pid] = pbar
         elif update is None:
             # not an update -> done
diff --git a/datalad/ui/progressbars.py b/datalad/ui/progressbars.py
index d5a3251322..d28ac793ff 100644
--- a/datalad/ui/progressbars.py
+++ b/datalad/ui/progressbars.py
@@ -215,7 +215,7 @@ class tqdmProgressBar(ProgressBarBase):
 
         def __init__(self, label='', fill_text=None,
                      total=None, unit='B', out=sys.stdout, leave=False,
-                     frontend=None, initial=0):
+                     frontend=None):
             """
 
             Parameters
@@ -228,12 +228,10 @@ def __init__(self, label='', fill_text=None,
             leave
             frontend: (None, 'ipython'), optional
               tqdm module to use.  Could be tqdm_notebook if under IPython
-            initial: (0, int), where to start the progress bar
             """
             super(tqdmProgressBar, self).__init__(label=label,
                                                   total=total,
-                                                  unit=unit,
-                                                  initial=initial)
+                                                  unit=unit)
 
             if frontend not in self._frontends:
                 raise ValueError(
```

</details>

I don't feel too strongly about this (and don't think it should hold up this PR if you or others disagree), but if we can avoid having _two_ spots that `intitial` can come in (i.e. `start` and `__init__`), I'd prefer to.

Reply:
@kyleam That makes sense, but (if I am not missing something) it ignores the (bad) situation of `ProgressBarBase` that still has the two slots for `initial`, and now leaves one of the original issues in that initially brought us here (provide `initial` to the constructor, only to have it be ignored, unless provided again to `start()`). I have no sufficient oversight to have a good concept of how progress bars are supposed to be used, so my POV is the one of a "victim" ;-)

Reply:
Sorry if I missing (at least part of) your point, but as I mentioned above, I think `ProgressBarBase.__init__` should lose its `initial` parameter.  The fact that its children don't have the same parameter is a sign that (1) things aren't wired up usefully and (2) it's not being used.  I'd be happy to see it done in this PR, though I don't want to push clean up onto @adswa.

Reply:
Too much time has been taken up by my comment though.  The `log_progress` API will stay the same either way, so I'd say we leave this as is and then any interested parties (maybe me :) can follow up with another PR at some point if they think it's worth it.

Reply:
I tried, and failed - I can't seem to wrap my head around this, sorry @kyleam :(